### PR TITLE
Initial docker command impl

### PR DESCRIPTION
**Pull request by JonnyWalker81**
Initial implementation of Docker commands.  It currently supports the `ps` and `images` Docker commands.  It should address Issue #619.

### CODE REVIEW

**Comment by jonathandturner** [comment]
Cool :) Glad to see someone starting on a docker command.

One suggestion I have here, if you want to wrap the commandline `docker` and give it a much more Nu-friendly way of working with it, is to add this as a plugin instead of a command.

From there, we can make it an optional feature of Nu (we did this with `tree` and `binaryview` as two other examples). Since not everyone will have docker installed, this gives people chance to optionally add the support if they want it.

**Comment by JonnyWalker81** [comment]
That makes sense, I will work on making it a plugin instead of a built in command.  Thank you for the feedback, I have enjoyed working on and contributing to this project.

**Comment by jonathandturner** [comment]
@JonnyWalker81 glad to hear it! 

We're working on our book for contributors, and there's a chapter on plugins if you're interested: https://github.com/nushell/contributor-book/blob/master/en/plugins.md

**Comment by JonnyWalker81** [comment]
I implemented a Plugin for the Docker command and removed the built-in Docker command I previously proposed.  The documentation in the contributor book was helpful, thank you.

File: src/cli.rs [hunk 1]
```diff
@@ -248,6 +248,7 @@ pub async fn cli() -> Result<(), Box<dyn Error>> {
             whole_stream_command(Next),
             whole_stream_command(Previous),
             whole_stream_command(Debug),
+            // whole_stream_command(Docker),
```
Comment (jonathandturner) [inline src/cli.rs]: You can just remove this line

File: src/commands.rs [hunk 2]
```diff
@@ -11,6 +11,7 @@ pub(crate) mod config;
 pub(crate) mod cp;
 pub(crate) mod date;
 pub(crate) mod debug;
+// pub(crate) mod docker;
```
Comment (jonathandturner) [inline src/commands.rs]: Same

File: src/commands.rs [hunk 3]
```diff
@@ -79,6 +80,7 @@ pub(crate) use config::Config;
 pub(crate) use cp::Cpy;
 pub(crate) use date::Date;
 pub(crate) use debug::Debug;
+// pub(crate) use docker::Docker;
```
Comment (jonathandturner) [inline src/commands.rs]: Same

File: src/cli.rs (hunk 1 above)
Comment (JonnyWalker81) [reply src/cli.rs]: I thought I had removed the commented out code, sorry about that.

**Comment by JonnyWalker81** [comment]
I added a `required-features` property in the Cargo.toml file similar to what I saw for other plugins, but I am not fully sure that is correct or if there is something else that is needed also.  I removed the commented out code as well.

**Comment by jonathandturner** [comment]
Great!

**Comment by iocanel** [comment]
How can I try this locally?

Tried `cargo build --features="docker` but it gave me compilation errors?
Do I miss something?
BTW, I am a complete rust noob.

**Pull request merged**